app/models/conversion_host.rb

```diff
@@ -16,6 +16,7 @@ class ConversionHost < ApplicationRecord
 
   validates :name, :presence => true
   validates :resource, :presence => true
+  validates :resource_id, :uniqueness => { :scope => :resource_type }
```

Comment:
@djberg96 interesting haven't seen this done before.  Did `validates :resource, :presence => true, :uniqueness => true` not work?

Reply:
@agrare I tried that originally, but rails did not like it. If you make that change and try to validate you will see something like this:

```
> c = ConversionHost.new(:resource => some_resource)
>
> c.valid?
Traceback (most recent call last):
       16: from activesupport (5.1.7) lib/active_support/callbacks.rb:601:in `block (2 levels) in default_terminator'
       15: from activesupport (5.1.7) lib/active_support/callbacks.rb:197:in `block (2 levels) in halting'
       14: from activesupport (5.1.7) lib/active_support/callbacks.rb:413:in `block in make_lambda'
       13: from activemodel (5.1.7) lib/active_model/validator.rb:147:in `validate'
       12: from activemodel (5.1.7) lib/active_model/validator.rb:147:in `each'
       11: from activemodel (5.1.7) lib/active_model/validator.rb:150:in `block in validate'
       10: from activerecord (5.1.7) lib/active_record/validations/uniqueness.rb:17:in `validate_each'
        9: from activerecord (5.1.7) lib/active_record/validations/uniqueness.rb:53:in `build_relation'
        8: from activerecord (5.1.7) lib/active_record/relation/query_methods.rb:612:in `where!'
        7: from activerecord (5.1.7) lib/active_record/relation/where_clause_factory.rb:19:in `build'
        6: from activerecord (5.1.7) lib/active_record/relation/where_clause_factory.rb:49:in `build_for_case_sensitive'
        5: from activerecord (5.1.7) lib/active_record/relation/where_clause_factory.rb:49:in `each'
        4: from activerecord (5.1.7) lib/active_record/relation/where_clause_factory.rb:52:in `block in build_for_case_sensitive'
        3: from activerecord (5.1.7) lib/active_record/reflection.rb:404:in `klass'
        2: from activerecord (5.1.7) lib/active_record/reflection.rb:408:in `compute_class'
        1: from activerecord (5.1.7) lib/active_record/inheritance.rb:166:in `compute_type'
NameError (uninitialized constant ConversionHost::Resource)
```

Reply:
and if you add `:polymorphic => true`

Reply:
On the validation line? You'll get:

`ArgumentError (Unknown validator: 'PolymorphicValidator')`

Reply:
:+1: this looks like a good workaround then

Reply:
Yeah, took some googling, was surprised the first approach didn't work. It probably *should* work, but here we are. :)